frontend/src/app/home/components/map/clickables/clickable-offices/clickable-offices.component.ts

```diff
@@ -0,0 +1,29 @@
+import { Component, OnChanges, Input, Output, EventEmitter, SimpleChanges } from '@angular/core';
+import { MainService } from './../../../../services/main.service';
+
+@Component({
+  selector: '[app-clickable-offices]',
+  templateUrl: './clickable-offices.component.html',
+  styleUrls: ['../clickable.scss', './clickable-offices.component.scss']
+})
+export class ClickableOfficesComponent implements OnChanges {
+  @Input() shiningOfficeId = [''];
+  @Output() officeId = new EventEmitter<string>();
+
+  constructor(
+    private MainService: MainService,
+  ) { }
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes.shiningOfficeId !== undefined) {
+      document.querySelectorAll(".shining").forEach(element => element.classList.remove("shining"));
+      if (this.shiningOfficeId[0] != '' && this.shiningOfficeId.length != 0) {
```

Comment:
ten warunek jest mało czytelny, bo żeby wiedzieć dlaczego sprawdza się `this.shiningOfficeId[0] != ''` trzeba wiedzieć, że początkowo jest tak zdefiniowane, a można sprawdzić np. `undefined`